File: Hashing/php/anagrams.php
<?php

declare(strict_types=1);

function anagrams(string $first, string $second): bool
{
    if (strlen($first) !== strlen($second)) {
        return false;
    }

    $counts = [];
    foreach (str_split($first) as $char) {
        $counts[$char] = ($counts[$char] ?? 0) + 1;
    }

    foreach (str_split($second) as $char) {
        if (!isset($counts[$char]) || $counts[$char] === 0) {
            return false;
        }
        $counts[$char]--;
    }

    return true;
}

File: Hashing/csharp/Anagrams.cs
using System.Collections.Generic;

namespace HashingSolutions;

public static class Anagrams
{
    public static bool Solve(string s1, string s2)
    {
        if (s1.Length != s2.Length)
        {
            return false;
        }

        var counts = new Dictionary<char, int>();
        foreach (char c in s1)
        {
            counts.TryGetValue(c, out int value);
            counts[c] = value + 1;
        }

        foreach (char c in s2)
        {
            if (!counts.TryGetValue(c, out int value) || value == 0)
            {
                return false;
            }
            counts[c] = value - 1;
        }

        return true;
    }
}
